### ISSUE TITLE
Godot randomly deletes stuff from .tscn files

### ISSUE THREAD

**Issue by ghost** [godotengine/godot#46027]
I have not found any issue mentioning the same bug. My apologies if there actually is one..

**Godot version:** 3.2.3

**OS/device including version:** Linux Mint 20.1 64-bit; the rest is probably irrelevant to the bug

**Issue description:** From time to time, Godot decides to delete stuff from a .tscn file of a scene I am working on. It doesn't show the change in the scene editor at all, which means the change might not be noticed by the user at all until he commits it and another person pulls the code. Fairly dangerous imho. The last time this happened, the undesired change was an icon property of a button being erased (as in it removed the icon ext_resources and removed the icon property from the button. The issue is very annoying, because it means I have to constantly check that Godot hasn't deleted something I don't want to delete, and fix it after him manually if he does it... then make sure to not open Godot as it would delete it again.

**Steps to reproduce:** None, happens completely randomly. It has happened several times at different places though, so it's not just a random glitch. In previous instances, I found that deleting the directories ~/.cache/godot and ~/.config/godot stopped this from happening, however this no longer the case.

**Minimal reproduction project:**

**Comment by Calinou** [comment]
Can you paste a diff of the scene when you encounter this bug? That is, use Git or another version control system to display the difference between the old working file and the new corrupted file.

**Comment by ofiala-a51** [comment]
```diff
diff --git a/game/game.tscn b/game/game.tscn
index 4a5be2c..f38f6b4 100644
--- a/game/game.tscn
+++ b/game/game.tscn
@@ -1,4 +1,4 @@
-[gd_scene load_steps=11 format=2]
+[gd_scene load_steps=10 format=2]
 
 [ext_resource path="res://game/game.gd" type="Script" id=1]
 [ext_resource path="res://game/environment/scrolling_background/scrolling_background.tscn" type="PackedScene" id=2]
@@ -7,7 +7,6 @@
 [ext_resource path="res://game/entities/mothership/mothership.tscn" type="PackedScene" id=5]
 [ext_resource path="res://game/ui/unit_button/unit_button.tscn" type="PackedScene" id=6]
 [ext_resource path="res://game/entities/basic_unit/assets/basic_unit.png" type="Texture" id=7]
-[ext_resource path="res://game/entities/dash_unit/assets/dash_unit.png" type="Texture" id=8]
 [ext_resource path="res://game/entities/tank_unit/assets/tank_unit.png" type="Texture" id=9]
 [ext_resource path="res://game/entities/mothership/assets/shield.png" type="Texture" id=10]
 
@@ -40,7 +39,6 @@ margin_left = 160.0
 margin_top = 940.0
 margin_right = 270.0
 margin_bottom = 1050.0
-icon = ExtResource( 8 )
 
 [node name="TankUnitButton" parent="HUD" instance=ExtResource( 6 )]
 margin_left = 290.0
```

**Comment by ofiala-a51** [comment]
Another case:
```diff
diff --git a/project.godot b/project.godot
index 04c4d3e..134cde2 100644
--- a/project.godot
+++ b/project.godot
@@ -199,7 +199,7 @@ _global_script_classes=[ {
 "language": "GDScript",
 "path": "res://game/environment/scrolling_background/scripts/scrolling_background.gd"
 }, {
-"base": "Healthbar",
+"base": "ProgressBar",
 "class": "Shieldbar",
 "language": "GDScript",
 "path": "res://game/ui/shieldbar/scripts/shieldbar.gd"
```

**Comment by ofiala-a51** [comment]
...and another one when moving stuff (it changed base class for a bunch of classes to `Reference`):
```diff
diff --git a/project.godot b/project.godot
index efcbbfa..c94b3b2 100644
--- a/project.godot
+++ b/project.godot
@@ -37,147 +37,147 @@ _global_script_classes=[ {
 "base": "Reference",
 "class": "GSAIAgentLocation",
 "language": "GDScript",
-"path": "res://libraries/godot-steering-ai-framework/GSAIAgentLocation.gd"
+"path": "res://libraries/gsai/GSAIAgentLocation.gd"
 }, {
-"base": "GSAISteeringBehavior",
+"base": "Reference",
 "class": "GSAIArrive",
 "language": "GDScript",
-"path": "res://libraries/godot-steering-ai-framework/Behaviors/GSAIArrive.gd"
+"path": "res://libraries/gsai/Behaviors/GSAIArrive.gd"
 }, {
-"base": "GSAIGroupBehavior",
+"base": "Reference",
 "class": "GSAIAvoidCollisions",
 "language": "GDScript",
-"path": "res://libraries/godot-steering-ai-framework/Behaviors/GSAIAvoidCollisions.gd"
+"path": "res://libraries/gsai/Behaviors/GSAIAvoidCollisions.gd"
 }, {
[...]
```

**Comment by Calinou** [comment]
@ofiala-a51 @jhonnystene Can you (or anyone else) still reproduce this bug in [Godot 3.4.2](https://godotengine.org/article/maintenance-release-godot-3-4-2) or any later release?

If yes, please ensure that an up-to-date Minimal Reproduction Project (MRP) is included in this report (a MRP is a zipped Godot project with the minimal elements necessary to reliably trigger the bug). You can upload ZIP files in an issue comment with a drag and drop.

**Comment by jhonnystene** [comment]
@Calinou I've still had it happen, but it seems to happen completely randomly, and there isn't any sort of reliable trigger that I've found so I can't really make an MRP.

**Comment by dpensky** [comment]
![image](https://user-images.githubusercontent.com/7250660/151379422-18167e3b-9ce0-4e01-93cd-d2654aadf655.png)

Not exactly deleting, but we constantly have that kind of difference in our projects like the above image. Our team uses 3 different operating systems, Windows, Linux and iOS, maybe it has something to do with it.

This issue also occurs in the export_presets.cfg file, where properties are saved in a different order.

**Comment by Calinou** [comment]
> ![image](https://user-images.githubusercontent.com/7250660/151379422-18167e3b-9ce0-4e01-93cd-d2654aadf655.png)
> 
> Not exactly deleting, but we constantly have that kind of difference in our projects like the above image. Our team uses 3 different operating systems, Windows, Linux and Mac, maybe it has something to do with it.
> 
> This issue also occurs in the export_presets.cfg file, where properties are saved in a different order.

This was reported elsewhere already, but I can't find the issue in question right now.

**Comment by AhmedBaari** [comment]
I encountered the same issue with my Godot project after making a minor change to the `main.gd` script. I added a line to start a tween which I forgot to do initially. So I went back and added `tween.start()`, but when I ran the project, it gave me a `parse error` on the `main.tscn` file. 

Upon further investigation, I found that all files with dependencies were empty, despite still existing in the file system (with no content). Interestingly, files I had created in the directory but not used in the project were untouched. For instance, 'player.gd' remained unchanged since it was created but never used in the project.

Not using Git in the project was a mistake on my part. I'm currently finding ways to recover my lost content, but no luck so far.

Godot Version: 3.3
Operating System: Windows 11

**Comment by akien-mga** [comment]
Godot 3.3 is no longer supported, so this can't be fixed in that version. You should update to Godot 3.5.3 to benefit from the latest fixes, which may very well have solved this since we haven't gotten new reports since 2021. Closing as very likely fixed in 3.5 / 4.x.
